sms_engine/models.py

```diff
@@ -106,3 +108,43 @@ class Meta:
 
     def __str__(self):
         return text_type(self.date)
+
+
+@python_2_unicode_compatible
+class Tag(models.Model):
+
+    KEY = 'sms-tag:%s'
+
+    name = models.CharField(max_length=255, unique=True)
+    sms = models.ManyToManyField(SMS, blank=True, related_name='sms')
+
+    def __str__(self):
+        return self.name
+
+    @classmethod
+    def get(cls, name, create=False):
+        slugified_name = slugify(name)
+
+        tag = cache.get(cls.KEY % slugified_name)
+        if tag:
+            return tag
+
+        tag = cls.objects.filter(name=name).first()
+        if tag:
+            cache.set(cls.KEY % (slugified_name), tag)
+        return tag
+
+    def save(self, *args, **kwargs):
+        # clear cache
+        if self.pk:
+            try:
+                old_name = Tag.objects.get(id=self.pk).name
+                cache.delete(Tag.KEY % (slugify(old_name)))
+            except Tag.DoesNotExist:
```

Comment:
Remove this `try` `except`, logically this should NEVER go to this `except` block, unless there are serious error